### DISCUSSION TITLE
No sound, no touchscreen calibration at startup

### DISCUSSION THREAD

**Discussion by Jean-Gabriel1982** [GrandOrgue/grandorgue#855] [General]
Hello,

when I start my pc (Linux Mint 20.2 Xfce), 2 times out of 3 I have no sound in GO or touch calibration (despite a calibration file also configured to start when the computer is started).

For that last point, I tried to put a delay after startup for the calibration file execution, but nothing change. 

I have to restart the pc several times (3 yesterday) so that everything returns to order without any further manipulation on my part ...

Thé day before yesterday, I’ve installed the Elo Touchscreen drivers [https://www.elotouch.fr/support/downloads#/category/346LYmeuAUEI4Qa0sSyiSa/os/4uN1G9fCeQckECOQACuyoW/legacy/current], but nothing changes, it is like I’ve installed nothing ..

This topic is different than de #750 one, as changing parameters in GO does not change, that's why I think it is not because of GO. 

Do you have an idea ?
Thank you very much in advance

**Comment by larspalo** [comment]
Do you have more than one soundcard? Even MIDI keyboards connected over USB can be listed as soundcards...

An old source of trouble was the order of initiation of different soundcards/devices that changed their internal names randomly - which in turn could surprise GO. There have been various efforts to make identification of the soundcards more stable, but I'd really suspect that the default soundcard might actually have changed when you have the issue.

**Reply by Jean-Gabriel1982** [reply]
@larspalo Thank you very much for your answer !

I have one sound card but connect my keyboard in usb. Yesterday evening all was good at first startup. 
I will ask my pc seller if I can "force" the pc to select my sound card at startup and with the same id ..

**Comment by oleg68** [comment]
@Jean-Gabriel1982 You can specify the exact soundcard it in the File->Settings->Audio Output.

I think the touchscreen problem is also caused by wrong order of initialising. There is a software solution for linux (udev rule + calibrating script).

**Reply by Jean-Gabriel1982** [reply]
@oleg68 ah I'll do that this evening :-)

Regarding touchscreen, I did xrandr and xinput map-to-output commands

I put my commands on a text file, configure it to be executable at startup.

The problem is when touchscreen works, in all case right clic does not work and when I click on a menu, I can't click on submenu 
For example in GO I can click on File, but I can't select "Settings" ..

**Reply by jason-barba** [reply]
I had the same problem (Submenus not working on touchscreen). I now us Ubuntu on Wayland without that problem. 

OS = Ubuntu 20.04LTS, touchscreen = Asus.

**Reply by Jean-Gabriel1982** [reply]
> I had the same problem (Submenus not working on touchscreen). I now us Ubuntu on Wayland without that problem. 
> 
> 
> 
> OS = Ubuntu 20.04LTS, touchscreen = Asus. 

Does that mean that to solve your touch screen problem, you changed your operating system and went to Ubuntu?

But I think it should be possible to properly configure my touch screens on the current operating system no!?

**Reply by jason-barba** [reply]
> Does that mean that to solve your touch screen problem, you changed your operating system and went to Ubuntu?

No, I have run GO on Ubuntu from the start. I just changed display server. By default Ubuntu 20.04 uses Xorg but on login you can change to Ubuntu on Wayland. 

> But I think it should be possible to properly configure my touch screens on the current operating system no!?

In theory, yes. But in my case it was enough to switch to Wayland so my knowledge in the matter stops there. 

See https://wiki.ubuntu.com/Wayland for more info.

**Comment by oleg68** [comment]
I have two touchscreens Philips, use them with Gnome/Xorg without any problems

**Comment by oleg68** [comment]
> But I think it should be possible to properly configure my touch screens on the current operating system no!?

Indead. Try to use different pairs Desktop Environment + Graphic Engine with your favorite linux distro.

My pair - Gnome+Xorg - works fine with Fedora. I think it will be fine with Mint too.

**Comment by Jean-Gabriel1982** [comment]
> > But I think it should be possible to properly configure my touch screens on the current operating system no!?
> 
> 
> 
> Indead. Try to use different pairs Desktop Environment + Graphic Engine with your favorite linux distro.
> 
How can I do ? 😋
I select the option at startup "start with Gnome environment" but nothing changes ...
> 
> My pair - Gnome+Xorg - works fine with Fedora. I think it will be fine with Mint too.

Maybe Xfce is a too light edition ?

**Comment by oleg68** [comment]
@Jean-Gabriel1982 I mananed to reproduce your issue with touchscreens on my Gnome+Xorg under Fedora. 

1. When I run GO installed on my comp or run it from tar.gz, the submenus work as expected.
2. When I run AppImage bundle of grandorgue then I can select a menu with touchscreen but can not select a submenu.

Seems the reason is wxWidgets. In (1) GO use wx installed on the system (wxGTK3-3.0.5.1), but when I run it from appimage, it uses wx library shipped with the bundle: 3.0.4+dfsg

Both mint and ubuntu also have wx 3.0.4.

Seems there is a touchscreen-related bug in wxWidgets 3.0.4 that has been fixed in 3.0.5.

But I found https://forums.wxwidgets.org/viewtopic.php?t=3611 where users have the same problem with an upstream version of wxWidgets without a solution. It requires more time for discovering.

**Reply by Jean-Gabriel1982** [reply]
> @oleg68 Isn't that just a driver and kernel version issue?  

Unprobably. It looks like xorg/gtk/wx issue. Seems it is fixed in fedora with some patch. I think this issue may be fixed also in mint. I need more time for discovering.

**Comment by Jean-Gabriel1982** [comment]
> @Jean-Gabriel1982 You can specify the exact soundcard it in the File->Settings->Audio Output.
> 
@oleg68 i think I have a problem 😂

Right now I start my computer : no sound, no touchscreen 

I go to File>Settings>Audio output and I have 

![image](https://user-images.githubusercontent.com/20370038/142269009-d40404ff-0473-4fbe-82a9-631f26c4c23c.jpeg)

**Comment by oleg68** [comment]
I can'not see any analog output. Maybe it is used by pulseaudio.

Try to stop pulseaudio with 

```
systemctl stop --user pulseaudio
```

then start GO and go to the settings. Which devices are available now?

**Reply by larspalo** [reply]
Usually it's impossible to stop pulseaudio as it will just respawn immediately. Instead one must prefix the GO startup command like so:

`pasuspender ./GrandOrgue`

**Comment by Jean-Gabriel1982** [comment]
> I can'not see any analog output. Maybe it is used by pulseaudio.
> 
> 
> 
> Try to stop pulseaudio with 
> 
> 
> 
> ```
> 
> systemctl stop --user pulseaudio
> 
> ```
> 
> 
> 
> then start GO and go to the settings. Which devices are available now?

With root mode ?
I have "no such file or folder"

**Reply by oleg68** [reply]
> > I can'not see any analog output. Maybe it is used by pulseaudio.
> > Try to stop pulseaudio with
> > ```
> > 
> > systemctl stop --user pulseaudio
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > then start GO and go to the settings. Which devices are available now?
> 
> With root mode ? I have "no such file or folder"

In user mode.

```
systemctl stop --user pulseaudio.socket
systemctl stop --user pulseaudio.service
```

If it solves the problem and you are using this comp only for GrandOrgue, you may disable pulseaudio permanently:

```
systemctl disable --user pulseaudio.socket
systemctl disable --user pulseaudio.service
```

If sometimes you use another applications requiring pulseaudio you may use pasuspender as Lars suggested.

And I suggest you to use 48000 sample rate because most of sample sets have this sample rate, otherwise resampling may reduce the quality of sound.

**Reply by Jean-Gabriel1982** [reply]
> In user mode.
> 
> 
> 
> ```
> 
> systemctl stop --user pulseaudio.socket
> 
> systemctl stop --user pulseaudio.service
> 

Doing this : when I start GO I have a warning "output not found pa:ALSA:default not found. No sound will be produced"

![image](https://user-images.githubusercontent.com/20370038/142274329-2592351e-bb95-4632-ac95-11a44bd5f419.jpeg)

> 
> If it solves the problem and you are using this comp only for GrandOrgue, you may disable pulseaudio permanently:
> 
> 
> 
> ```
> 
> systemctl disable --user pulseaudio.socket
> 
> systemctl disable --user pulseaudio.service
> 
> ```
> 
> 
> 
> If sometimes you use another applications requiring pulseaudio you may use pasuspender as Lars suggested.

On this computer I only use GO.

**Reply by oleg68** [reply]
> Doing this : when I start GO I have a warning "output not found pa:ALSA:default not found. No sound will be produced"

Because PulseAudio provided the "default" device that was not safe across reboots. Fix another device in File->Settings->Audio and this will disappear.

**Reply by oleg68** [reply]
Yes, the new devices appeared:

``Pa: ALSA: HD-Audio Generic ALC1120 Analog (hw:2,0)`` and ``Rt: alsa: hw:HD-Audio Generic,0``. They are the same devices with two different sound ports. I usually prefer Rt, but Pa also will work. Please select one of them and check.

If it works, disable pulseaudio permanently. Reboot the comp and test.

**Reply by Jean-Gabriel1982** [reply]
> > Doing this : when I start GO I have a warning "output not found pa:ALSA:default not found. No sound will be produced"
> 
> 
> 
> Because PulseAudio provided the "default" device that was not safe across reboots. Fix another device in File->Settings->Audio and this will disappear.
> 
> 

Ok. I tried different options in Audio output and sound came back when I've selected 
>pa:ALSA: HD-Audio Generic:ALC1220 Analog (hw:2,0)

**Reply by larspalo** [reply]
> If it works, disable pulseaudio permanently. Reboot the comp and test.

I'd suggest that you do this only if you **only** use GO on that computer and nothing else. You'll more than likely have issues with other applications sound outputs.

**Reply by oleg68** [reply]
> I'd suggest that you do this only if you only use GO on that computer and nothing else. You'll more than likely have issues with other applications sound outputs.

I agree.

**Reply by larspalo** [reply]
But, yes, I always do the same with pasuspender to get direct access to the soundcard when I'm using GO!

**Reply by Jean-Gabriel1982** [reply]
Yes I've bought this computer only for GO. I think I won't use it for something else. 

So I've disabled pulseaudio, and reboot the computer.. at startup 😭

![image](https://user-images.githubusercontent.com/20370038/142276383-7f144dfd-3e4f-46ea-b0d2-2bfc8137a555.jpeg)

**Comment by oleg68** [comment]
May be pulseaudio is still alive?

**Comment by Jean-Gabriel1982** [comment]
> May be pulseaudio is still alive?

How can I kill it ? 🪓 🔪 💣 
😂

**Reply by oleg68** [reply]
systemctl stop ...

**Reply by larspalo** [reply]
I'd suggest you change the program starter line for GO to include (be prefixed by) pasuspender.

**Reply by Jean-Gabriel1982** [reply]
I did that

> systemctl stop --user pulseaudio.socket
systemctl stop --user pulseaudio.service

And after I did that

>systemctl disable --user pulseaudio.socket
systemctl disable --user pulseaudio.se

I reboot the pc and I have this warning ..

When I click "valider" the organ loads and the sound is ok. But at each startup I have the warning

**Reply by Jean-Gabriel1982** [reply]
> I'd suggest you change the program starter line for GO to include (be prefixed by) pasuspender.

Ok i put pasuspender /bin/GrandOrgue and rebooted the computer. 
At GO startup, I have no warning but no sound too ....

ÉDIT : when I put off pasuspender of startup, I have the warning but sound works .. it might be a coincidence ...

**Reply by larspalo** [reply]
Does anything change with the listed audio devices when you start with pasuspender?

**Reply by Jean-Gabriel1982** [reply]
> Does anything change with the listed audio devices when you start with pasuspender?

I think not

![image](https://user-images.githubusercontent.com/20370038/142283573-1d3a9910-648c-4063-9900-44a23273f68c.jpeg)

**Comment by oleg68** [comment]
@Jean-Gabriel1982 Are you still having any problems with the sound after reboot now?

**Reply by Jean-Gabriel1982** [reply]
@oleg68 As the concern arises randomly I cannot tell you precisely.  

this morning the organ turned on without concern, the sound was present and the Rt ALSA output was correctly selected (in the sound ports I unchecked "Pa")

The touchscreen was correct too. 

Maybe before leaving home for work I'll test again and tell you if it works fine.

**Reply by Jean-Gabriel1982** [reply]
@oleg68 Ive just started the computer ... no sound, no touchscreen... snif

ÉDIT : i reboot the computer, the result is same. 

ÉDIT 2: i quit GO and start again without rebooting computer : I have the sound but not the touchscreen

**Comment by oleg68** [comment]
> Ive just started the computer ... no sound, no touchscreen... snif

Lets not solve two problems at the same time and at first achieve a durable working of sound.

1. Please return the pulseauio on the boot up

    ```
    systemctl enable --user pulseaudio.service
    systemctl enable --user pulseaudio.socket
    systemctl start --user pulseaudio.service
    systemctl start --user pulseaudio.socket
    ```
2. Remove pasuspender from starting GO
3. Start GO. Set the audio device to ``Rt: alsa: default`` and check that it works.
4. Restore pasuspender and start GO with it. You will reveive the error that ``Rt: alsa: default`` doesn't exist. Set the sample rate to 48000, select ``Rt: alsa: hw:HD-Audio Generic,0`` audio device and check that it works.
5. Reboot your comp and check that sound works.

If something goes wrong tell here what step has failed.

**Reply by Jean-Gabriel1982** [reply]
Yes ! I'll do that this evening when I come back home ;-)

**Comment by Jean-Gabriel1982** [comment]
> 1. Please return the pulseauio on the boot up

>     ```
>     systemctl enable --user pulseaudio.service
> 
>     systemctl enable --user pulseaudio.socket
> 
>     systemctl start --user pulseaudio.service
> 
>     systemctl start --user pulseaudio.socket
> 
>     ```

Done

> 2. Remove pasuspender from starting GO
> 

Done

> 3. Start GO. Set the audio device to ``Rt: alsa: default`` and check that it works.
> 

It works

> 4. Restore pasuspender and start GO with it. You will reveive the error that ``Rt: alsa: default`` doesn't exist. Set the sample rate to 48000, select ``Rt: alsa: hw:HD-Audio Generic,0`` audio device and check that it works.
> 

By restoring pasuspender you mean in Startup program ? 

Pasuspender /bin/GrandOrgue 

When I reboot the computer I don't have any error message. There is no sound. 

When I set the sample rate to 48000 and select the audio device, GO crashed I could not click on anything at all neither the stops not the menus. 

I forced GO to close and reopened it. the sound worked but the sample rate was back to 44100 and RT ALSA default as audio device. 

I put sample rate to 48000 and there is still sound. I wanted to change the audio device but in the options I don't have audio generic 0 anymore. I have audio generic 1,3,7,8 or 9
I let Rt ALSA default, save parameters and reboot the computer


> 5. Reboot your comp and check that sound works.
> 
Rebooting done. No sound. 
RtALSA hwHD Audio Generic 0 came back. 
I select it and click "done"
GO crashes I can't click on anything ... I force GO to close
> 
> If something goes wrong tell here what step has failed.

**Reply by oleg68** [reply]
> By restoring pasuspender you mean in Startup program ?
> 
> Pasuspender /bin/GrandOrgue

No. In the terminal window in user mode. Let at first we get working config without reboot then we will achieve the same config working across reboots.

**Comment by Jean-Gabriel1982** [comment]
I've installed other organs for testing #707 but I have no sound. 

Sample rate is 48000 and if I want to put 44100, GO crashes. 
If I just want to change the Audio device, GO crashes
If I want to select or de select Sound Ports, GO crashes 

I tried to click on Panic but same result. 

I think Sample rate is responsible...

**Reply by Jean-Gabriel1982** [reply]
This morning, I've just remove "pasuspender" from path of GO in Startup program. 

At startup, all is good this morning, I've changed sample rate to 44100 without problem. 

The audio output is RtALSA default. 
The sound port "Pa" is not selected

**Reply by larspalo** [reply]
Just to set the background: others milage may vary, this is just my general preferences, and I do use other setups of device/drivers from time to time when I need to. And yes, I do have multiple soundcards on the computer.

Do you have any real reason for preferring Rt other than Olegs word? Just for testing, enable Pa instead and disable all Rt - both jack and alsa and also disable Jack native. Select anything from the list as device. Close GO and start GO again from a terminal with:

`pasuspender /path/to/GrandOrgue`

Select the audio device that's most close to the actual hardware (in my case it's identified with Pa: ALSA: io|2: USB Audio (hw:2,0)). In my particular setup Rt cannot identify this device at all - with or without pulseaudio.

See if you get any sound with this strategy.

**Reply by oleg68** [reply]
> I've installed other organs for testing #707 but I have no sound.
> 
> Sample rate is 48000 and if I want to put 44100, GO crashes. If I just want to change the Audio device, GO crashes If I want to select or de select Sound Ports, GO crashes

Crash is a vary bad thing. If you manage to reproduce it please create an issue with step-by-step descriprion how to reproduce it. It must be fixed in all cases

**Reply by Jean-Gabriel1982** [reply]
@larspalo 
I close GO and start again in the terminal with the command you gave.
There is no sound, and the terminal returns this

`pasuspender /bin/GrandOrgue
ALSA lib pcm_dsnoop.c:641:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dmix.c:1089:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm.c:2642:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2642:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2642:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_dmix.c:1089:(snd_pcm_dmix_open) unable to open slave
Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiInJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiInJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiOutJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiOutJack::initialize: JACK server not running?
`

**Comment by Jean-Gabriel1982** [comment]
> Just to set the background: others milage may vary, this is just my general preferences, and I do use other setups of device/drivers from time to time when I need to. And yes, I do have multiple soundcards on the computer.
> 
> Do you have any real reason for preferring Rt other than Olegs word? Just for testing, enable Pa instead and disable all Rt - both jack and alsa and also disable Jack native. Select anything from the list as device. Close GO and start GO again from a terminal with:
> 
> `pasuspender /path/to/GrandOrgue`
> 
> Select the audio device that's most close to the actual hardware (in my case it's identified with Pa: ALSA: io|2: USB Audio (hw:2,0)). In my particular setup Rt cannot identify this device at all - with or without pulseaudio.
> 
> See if you get any sound with this strategy.

@larspalo I'm not fond of Rt or Pa .. i just want everything works well and always without be afraid at each reboot, to ask myself if i will have sound or not ;-)

This evening, i start my computer, everything was good.
I disable all Rt and enable Pa. I choose Pa Alsa default and it works ... but i haven't rebooted the computer.
I chose Pa Alsa default, but there are other options ... but i don't know what to choose really ..

@oleg68 
I tried to reproduce the crash but no way. Maybe i have to put pasuspender in my path when computer starts .. I'll try.
But as i put pasuspender in a wrong place ..

**Comment by oleg68** [comment]
@Jean-Gabriel1982 

> I close GO and start again in the terminal with the command you gave.
There is no sound, and the terminal returns this ``pasuspender /bin/GrandOrgue``

OK, seems, pasuspender doesn't work as expected. Lets find the reason.

1. Temporary disable pulseaudio
    
    ```
    systemctl stop --user pulseaudio.socket
    systemctl stop --user pulseaudio.service
    ```

 2. Start GO without pasuspender, go to Settings->Audio and select any ``hw:HD-Audio Generic,0`` device ether with Pa or with Rt (Rt produces less of spam in the terminal window because it does not probe jack)
 3. Check that sound works
 4. Exit from GO. The settings remember the ``hw:HD-Audio Generic,0`` device
 5. Reenable pulseaudio

    ```
    systemctl start --user pulseaudio.service
    systemctl start --user pulseaudio.socket
    ```
6. Try to start GO without pasuspender. The error will be displayed and the settings dialog will be opened. **Exit with Cancel** from Settings. **Do not press OK** otherwise the setting file will be overwritten.
7. Check that sound **is not working** and pressing to `Panic` does not help
8. Then start GO from the terminal with pasuspender. If GO still complains about the device unavailable, **do not try to select another device** just press OK and exit from settings
10. Check, wether sound is working?
11. If it is still not working, try to press `Panic` and check the sound again
12. If it is still not working, wait 10 seconds or more, press `Panic` and check the sound again

What happens on each step?

**Reply by Jean-Gabriel1982** [reply]
@oleg68 

Hello so this morning I started the computer.  Everything was correct.  The sample rate was at 48000, all audio options were selected and the output was Rt. 

I lowered the sample rate to 44100 (I found a very slight note delay when playing, slight but unsettling). 

I closed GO and reopened it: nickel. 
I restarted the computer: nickel. 
I turned the computer off and on again: nickel.  

But just before I left for work, I turned the computer back on and there… No sound, no touchscreen. 
Is the sound related to the touch?  I don’t know .. is the problem with starting the computer? 

And yet I changed absolutely nothing in the settings between the moment it worked fine and the moment when nothing worked ...

I'll do your steps this evening.

**Comment by Jean-Gabriel1982** [comment]
@oleg68 
> 
> > I close GO and start again in the terminal with the command you gave.
> > There is no sound, and the terminal returns this `pasuspender /bin/GrandOrgue`
> 
> OK, seems, pasuspender doesn't work as expected. Lets find the reason.
> 
>     1. Temporary disable pulseaudio
>        ```
>        systemctl stop --user pulseaudio.socket
>        systemctl stop --user pulseaudio.service
>        ```
Done

>     2. Start GO without pasuspender, go to Settings->Audio and select any `hw:HD-Audio Generic,0` device ether with Pa or with Rt (Rt produces less of spam in the terminal window because it does not probe jack)

Done I have just Rt choice. With Pa, i don't have "Generic,0" choice.
 
>     3. Check that sound works

It works.
By the way, I had a little latency between pressing the key and when the sound came out, and doing your first two steps removed that latency .. it was very unpleasant and now really cool :)
> 
>     4. Exit from GO. The settings remember the `hw:HD-Audio Generic,0` device
> 
>     5. Reenable pulseaudio
>        ```
>        systemctl start --user pulseaudio.service
>        systemctl start --user pulseaudio.socket
>        ```
Done

>     6. Try to start GO without pasuspender. The error will be displayed and the settings dialog will be opened. **Exit with Cancel** from Settings. **Do not press OK** otherwise the setting file will be overwritten.
> 
Without pasuspender, GO opens normally. I don't have any error and the sound is still here
`Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiInJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiInJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiOutJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiOutJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiInJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiOutJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiInJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiOutJack::initialize: JACK server not running?
`

>     7. Check that sound **is not working** and pressing to `Panic` does not help
> 
As sound works even the Step 6, i have not pressed "Panic" button.

>     8. Then start GO from the terminal with pasuspender. If GO still complains about the device unavailable, **do not try to select another device** just press OK and exit from settings
> 
Done. I don't have any message with or without pasuspender.
With pasuspender, GO starts with sound ok and terminal gives

`Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiInJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiInJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiOutJack::initialize: JACK server not running?

Cannot connect to server socket err = Aucun fichier ou dossier de ce type
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

MidiOutJack::initialize: JACK server not running?`


>     9. Check, wether sound is working?
> 
>     10. If it is still not working, try to press `Panic` and check the sound again
> 
>     11. If it is still not working, wait 10 seconds or more, press `Panic` and check the sound again
> 
> 
> What happens on each step?

**Comment by oleg68** [comment]
> Without pasuspender, GO opens normally. I don't have any error and the sound is still here

It means that pulseaudio doesn't work as expected. It should lock the audio card againsg GO to use it.

Lets repeat the step 5 and then open the system audio settings: 
1. please click on the speaker icon on the bottom panel and press ``Audio Mixer->Output Devices``. 
2. Try to move the volume slider. Do you hear anything?
3. What device present in the Port?
4. Select ``Analog Output`` port and repeat moving the volume slider.
5. If your succeed getting the audio working, repeat the previous list from the step 6 (running GO without pasuspender; the sound shouldn't work at this step and should work later with pasuspender).

**Comment by Jean-Gabriel1982** [comment]
> 



> > Without pasuspender, GO opens normally. I don't have any error and the sound is still here
> 
> It means that pulseaudio doesn't work as expected. It should lock the audio card againsg GO to use it.
> 
> Lets repeat the step 5 and then open the system audio settings:
> 
>     1. please click on the speaker icon on the bottom panel and press `Audio Mixer->Output Devices`.
> 
>     2. Try to move the volume slider. Do you hear anything?
> 
I don't hear anything

>     3. What device present in the Port?
> 
It is : Family 17h (Models 10-1fh) HD Audio Controler Stereo Analogic
In "Port" : line out (plugged in)

>     4. Select `Analog Output` port and repeat moving the volume slider.
> 
I don't have. 
Port : Line out (plugged in) or headphones (unplugged)

>     5. If your succeed getting the audio working, repeat the previous list from the step 6 (running GO without pasuspender; the sound shouldn't work at this step and should work later with pasuspender).

**Reply by oleg68** [reply]
Select Line out and check the volume slider

**Reply by Jean-Gabriel1982** [reply]
No sound except  little "wind" when slider is 100%

**Reply by oleg68** [reply]
It is Ok. Return to the step 6 from the prev. list

**Reply by Jean-Gabriel1982** [reply]
Ok 

6/ Ok i have the error. I proceed as requested

7/ Panic gives me 
`Error RtAudio: RtApiAlsa::probeDeviceOpen: pcm device (hw:2,0) won't open for output`

6/7/ i have no sound

8/ ok i haven't the Error

9/ sound is working

**Reply by oleg68** [reply]
Ok. now sound does not work without pasuspender and works with pasuspender.

Please make the start up setting that GO would start with pasuspender and reboot the pc. What happens?

**Reply by Jean-Gabriel1982** [reply]
So in "session and start" program, in the path of GO I add well pasuspender?
pasuspender /bin/GrandOrgue ?

**Reply by oleg68** [reply]
Yes, you do

**Reply by Jean-Gabriel1982** [reply]
GO started well, with the sound :)

**Reply by Jean-Gabriel1982** [reply]
mmm.. euh .. i shut off the computer, and wait some minutes. I just start it, Go, starts, but i don't have any sound..

**Reply by oleg68** [reply]
Congratulations!
Now I suggest you
1. To save the GrandOrgueConfig file
2. To try 48000 for better sound quality
3. To try 512 or even 256 samples per buffer for reducing latency

**Reply by Jean-Gabriel1982** [reply]
I save in cmb file i think it is that :)
I put 48000 and 512, but as i wrote just before, i started the computer and there is no sound again :(

- EDIT : I closed GO and start from the terminal with pasuspender. I have the sound.
So the problem seems to be at startup pfffff

- ÉDIT 2 : problem probably solved by adding `sh -c "sleep ` command in startup program before pasuspender. I put 5 seconds (an eternity as I was so happy the organ was ready in 35 seconds 😂).

So GO starts 5 seconds after computer startup and it seems to work. 
Case to follow ....

I'll try to do the same with touchscreen

**Comment by oleg68** [comment]
@Jean-Gabriel1982 Yes, the root cause is clear: the sound can works only after pulseaudio has initialised the sound card properly. And the start program should wait for it. Otherwise pasuspender stops pulseaudio and the sound card remains unitialised.

May be some better way exists for doing init rather than sleep, but I don't know it. If the sleep workaround is acceptable for you, lets go to the touchscreens.

1. Which problems with touchscreens do you have except the submenus are not selectable?
2. Do you always have problems with touchscreens or only sometimes after reboot?
3. Does running the calibration script manually solve the problems?

**Comment by Jean-Gabriel1982** [comment]
> May be some better way exists for doing init rather than sleep, but I don't know it. If the sleep workaround is acceptable for you, lets go to the touchscreens.
> 
I haven't found any other means than sleep, by searching the net a bit, but if there is a better way, or at least a safer way, I'm in favor.

Afterwards, if the sleep method works, that's fine .. as long as it works and I don't turn on the computer with fear in my stomach :)

But it seems that on Mint xfce there is no access to the timer, as on other systems (Mint Cinnamon for example has a timer option for the program at startup)

>     1. Which problems with touchscreens do you have except the submenus are not selectable?
> 
Often, when starting or restarting the computer, the touch is not calibrated. That is to say that when I press on a register of the organ, the mouse clicks 10-15 cm to the left or to the right

>     2. Do you always have problems with touchscreens or only sometimes after reboot?
> 
Sometimes the problem is very random
This morning I started the organ and everything was good.

I restarted it twice, and both times the touch was no longer calibrated.
Already, this means that the "sleep" method for calibration is useless.

>     3. Does running the calibration script manually solve the problems?

It seems not.
I run on my terminal the two commands 
xinput map-to-output 9 HDMI-A-1
xinput map-to-output 11 HDMI-A-0

But there is no change

**Comment by oleg68** [comment]
@Jean-Gabriel1982 We have already started a topic about calibrating touchscreens #736. Lets continue there.
